proc/groupby.go

```diff
@@ -303,67 +342,71 @@ func (g *GroupByAggregator) recordsForTable(table map[string]*GroupByRow) []*zso
 	}
 	sort.Strings(keys)
 	var recs []*zson.Record
-Keys:
-	for _, k := range keys {
-		row := table[k]
-		var vals [][]byte
-		if g.TimeBinDuration > 0 {
-			vals = append(vals, []byte(row.ts.StringFloat()))
-		}
-		for _, v := range row.keyVals {
-			vals = append(vals, v)
-		}
-		for _, red := range row.columns.Reducers {
-			v, err := zson.ZvalFromZeekString(nil, reducer.Result(red).String())
-			if err != nil {
-				g.logger.Error("zson.ZvalFromZeekString failed", zap.Error(err))
-				continue Keys
+
+	for _, typeCol := range g.typeCols {
```

Comment:
I don't understand the purpose of the outer loop (and the typeMatch check in the inner loop)?

Reply:
Adding this comment...
```
	// We make multiple passes over the table, one for each unique type 
	// vector or row keys, the in each pass we output as a batch all the 
	// records that have this unique type (i.e., the same descriptor).
	// This allows variations in the type if the join-key to be tracked 
	// separately and output with their correct descriptor.
```

Reply:
"then" in each pass

Reply:
I think the fact that keys begin with type information and those get sorted makes the whole `typeCols` thing unnecessary?

Reply:
Yes good observation but the loop would take a bit more reworking so let's do this improvement in a subsequent PR.